Makefile

```diff
@@ -0,0 +1,47 @@
+BINDIR=/usr/local/bin
+UNITDIR=/etc/systemd/system
+SYSCONFDIR=/usr/local/etc
+DEFAULTDIR=$(SYSCONFDIR)/default
+DESTDIR=
+PROGNAME=alertmanager_matrix
+
+all: bin/$(PROGNAME) $(PROGNAME).service
+
+ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
```

Comment:
I don't think this variable should be needed? Execution of a Makefile should be in its directory afaik.

Reply:
That's an assumption that will cause data loss to someone in the future.  `make -F ../Makefile` comes to mind.  This specific construct using `ROOT_DIR` comes from direct and painful experience.

Check first answer here: https://unix.stackexchange.com/questions/150786/why-current-directory-doesnt-change-in-makefile

Reply:
Well, `make -F ../Makefile` is the wrong command for that. You should be using `make -C ..`

Reply:
And that is something you and I know, but others may not, hence the potential data loss.

Can we proceed as-is?  Is there any practical harm in leaving the variable there?

Reply:
The harm is that it negatively impacts readability. Makefiles are complicated enough as is, and I would like to have it as clean readable as possible.